File: backend/src/main/java/com/collab/service/AuthService.java
package com.collab.service;

import com.collab.model.dto.AuthRequest;
import com.collab.model.dto.AuthResponse;
import com.collab.model.entity.User;
import com.collab.repository.UserRepository;
import com.collab.util.JwtUtil;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthService {

    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtUtil jwtUtil;

    public AuthResponse register(AuthRequest request) {
        if (userRepository.existsByUsername(request.getUsername())) {
            throw new IllegalArgumentException("用户名已存在");
        }
        if (request.getEmail() == null || request.getEmail().isBlank()) {
            throw new IllegalArgumentException("邮箱不能为空");
        }
        if (userRepository.existsByEmail(request.getEmail())) {
            throw new IllegalArgumentException("邮箱已存在");
        }

        User user = userRepository.save(User.builder()
                .username(request.getUsername())
                .email(request.getEmail())
                .passwordHash(passwordEncoder.encode(request.getPassword()))
                .avatarUrl("https://ui-avatars.com/api/?name=" + request.getUsername() + "&background=random")
                .build());

        return buildResponse(user);
    }

    public AuthResponse login(AuthRequest request) {
        User user = userRepository.findByUsername(request.getUsername())
                .orElseThrow(() -> new IllegalArgumentException("用户不存在"));

        if (!passwordEncoder.matches(request.getPassword(), user.getPasswordHash())) {
            throw new IllegalArgumentException("密码错误");
        }

        return buildResponse(user);
    }

    private AuthResponse buildResponse(User user) {
        return AuthResponse.builder()
                .token(jwtUtil.generateToken(user.getUsername(), user.getId()))
                .userId(user.getId())
                .username(user.getUsername())
                .email(user.getEmail())
                .avatarUrl(user.getAvatarUrl())
                .build();
    }
}

File: python-backend/app/services/auth_service.py
import bcrypt

from app.models.user import User
from app.utils.jwt import create_token


class AuthService:
    def __init__(self, db):
        self.db = db

    def register(self, username: str, password: str, email: str | None = None) -> dict:
        if self.db.query(User).filter(User.username == username).first():
            raise ValueError("用户名已存在")

        email = email or f"{username}@example.com"
        if self.db.query(User).filter(User.email == email).first():
            raise ValueError("邮箱已存在")

        pw_hash = bcrypt.hashpw(password.encode("utf-8"), bcrypt.gensalt()).decode("utf-8")
        user = User(
            username=username,
            email=email,
            password_hash=pw_hash,
            avatar_url=f"https://ui-avatars.com/api/?name={username}&background=random",
        )
        self.db.add(user)
        self.db.commit()
        self.db.refresh(user)
        return self._build_response(user)

    def login(self, username: str, password: str) -> dict:
        user = self.db.query(User).filter(User.username == username).first()
        if not user:
            raise ValueError("用户不存在")

        if not bcrypt.checkpw(password.encode("utf-8"), user.password_hash.encode("utf-8")):
            raise ValueError("密码错误")

        return self._build_response(user)

    def _build_response(self, user: User) -> dict:
        return {
            "token": create_token(user.username, user.id),
            "userId": user.id,
            "username": user.username,
            "email": user.email,
            "avatarUrl": user.avatar_url or "",
        }
